packages/components/src/components/buttons/buttonStyleUtils.ts

```diff
@@ -11,48 +12,30 @@ export type ButtonVariant =
-
-    // Button lies always on elevation so for example Button that lies has elevation 0, lies on elevation -1.
-    // This is why the values here a shifted by 1.
-    0: 'interactionBackgroundTertiaryDefaultNormalOnElevationNegative',
-    1: 'interactionBackgroundTertiaryDefaultNormalOnElevation0',
-    2: 'interactionBackgroundTertiaryDefaultNormalOnElevation1',
-    3: 'interactionBackgroundTertiaryDefaultNormalOnElevation2',
-};
-
-export const mapElevationToBackgroundTertiaryHover: Record<Elevation, Color> = {
-    '-1': 'interactionBackgroundTertiaryDefaultHoverOnElevation3', // This is here just to satisfy Typescript mapping
-
-    // Button lies always on elevation so for example Button that lies has elevation 0, lies on elevation -1.
-    // This is why the values here a shifted by 1.
-    0: 'interactionBackgroundTertiaryDefaultHoverOnElevationNegative',
-    1: 'interactionBackgroundTertiaryDefaultHoverOnElevation0',
-    2: 'interactionBackgroundTertiaryDefaultHoverOnElevation1',
-    3: 'interactionBackgroundTertiaryDefaultHoverOnElevation2',
-};
-
-export const mapElevationToBackgroundInfoNormal: Record<Elevation, Color> = {
-    '-1': 'interactionBackgroundInfoDefaultHoverOnElevation3', // For example left menu is negative elevation
-
-    // Button lies always on elevation so for example Button that lies has elevation 0, lies on elevation -1.
-    // This is why the values here a shifted by 1.
-    0: 'interactionBackgroundInfoDefaultNormalOnElevationNegative',
-    1: 'interactionBackgroundInfoDefaultNormalOnElevation0',
-    2: 'interactionBackgroundInfoDefaultNormalOnElevation1',
-    3: 'interactionBackgroundInfoDefaultNormalOnElevation2',
-};
-
-export const mapElevationToBackgroundInfoHover: Record<Elevation, Color> = {
-    '-1': 'interactionBackgroundInfoDefaultHoverOnElevation3', // This is here just to satisfy Typescript mapping
+const mapElevationToButtonBackground = ({
+    elevation,
+    variant,
+    state,
+}: {
+    elevation: Elevation;
+    variant: 'destructive' | 'tertiary' | 'info' | 'warning';
+    state: 'normal' | 'hover';
+}): Color => {
+    const capitalizedVariant = capitalizeFirstLetter(variant);
+    const capitalizedState = capitalizeFirstLetter(state);
+
+    const map: Record<Elevation, Color> = {
+        '-1': `interactionBackground${capitalizedVariant}DefaultHoverOnElevation3`, // For example left menu is negative elevation
+
+        // Button lies always on elevation so for example Button that lies has elevation 0, lies on elevation -1.
+        // This is why the values here a shifted by 1.
+        0: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevationNegative`,
+        1: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation0`,
+        2: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation1`,
+        3: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation2`,
```

Comment:
How about `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation${elevation}` here and in other places?

Reply:
Somethime it works like this in `mapElevationToBackground` however this has of 1 shift so its not that simple and I think its more readable like this.